amd/src/annoto.js

```diff
@@ -183,6 +196,112 @@ define([
             } else {
                 log.warn('Annoto not loaded');
             }
-        }
+        },
+
+        annotoReady: function(api) {
+            // Api is the API to be used after Annoot is setup
+            // It can be used for SSO auth.
+            // https://github.com/Annoto/widget-api/blob/master/lib/index.d.ts#L12.
+            var pluginApi = api;
+
+            /*
+             * This should be essentially same as here:
+             * https://github.com/Annoto/moodle-local_annoto/blob/master/amd/src/annoto.js#L167
+             * below is simplified version.
+             */
+            pluginApi.auth(this.params.userToken).catch(function() {
+                log.error('Annoto SSO auth error');
+            });
+
+        },
+
+        annotoSetup: function(config) {
+            /*
+             *  config will contain the annoto widget configuration.
+             * This hook provides a chance to modify the configuration if required.
+             * Below we use this chance to attach the ssoAuthRequestHandle and mediaDetails hooks.
+             * https://github.com/Annoto/widget-api/blob/master/lib/config.d.ts#L128
+             *
+             * For the moodle plugin this the place to make all the configuration as done here
+             * https://github.com/Annoto/moodle-local_annoto/blob/master/amd/src/annoto.js#L118
+             *
+             * NOTICE: config is already setup by the Kaltura Annoto plugin,
+             * so we need only to override the required configuration, such as
+             * clientId, features, etc. DO NOT CHANGE THE PLAYER TYPE OR PLAYER ELEMENT CONFIG.
+            */
+            var params = this.params;
+
+            var innerAlignPlayers = ['h5p'];
+            var horizontalAlign = 'element_edge';
+            if (!params.widgetOverlay || params.widgetOverlay === 'auto') {
+                horizontalAlign = (innerAlignPlayers.indexOf(params.playerType) !== -1) ? 'inner' : 'element_edge';
+            } else if (params.widgetOverlay === 'inner') {
+                horizontalAlign = 'inner';
+            }
+
+            var configSrc = {
+                clientId: params.clientId,
+                position: params.position,
+                features: {
+                    tabs: params.featureTab,
+                    cta: params.featureCTA,
+                },
+                width: {
+                    max: (horizontalAlign === 'inner') ? 320 : 360,
+                },
+                align: {
+                    vertical: params.alignVertical,
+                    horizontal: horizontalAlign,
```

Comment:
@ilyaDymshits remove the `horizontalAlign` config (see the notes above). keep only the `align.vertical`